### ISSUE TITLE
"Did not expect successive traversals" when parsing 2 sibling html comments

### ISSUE THREAD

**Issue by bago** [cheeriojs/cheerio#2500]
Since 1.0.0-rc.6 cheerio give me fatal exception while dealing with fragments having comments:

```javascript
    var $ = require('cheerio').load("whatever");
    const a = $('<!-- comment --><!-- anothercomment -->');
```

```text
Error: Did not expect successive traversals.
    at ensureNotTraversal (C:\Lab\Void\projects\mosaico\node_modules\css-what\lib\commonjs\parse.js:137:19)
    at addTraversal (C:\Lab\Void\projects\mosaico\node_modules\css-what\lib\commonjs\parse.js:146:9)
    at parseSelector (C:\Lab\Void\projects\mosaico\node_modules\css-what\lib\commonjs\parse.js:204:17)
    at parse (C:\Lab\Void\projects\mosaico\node_modules\css-what\lib\commonjs\parse.js:83:20)
    at Object.select (C:\Lab\Void\projects\mosaico\node_modules\cheerio-select\lib\index.js:191:65)
    at LoadedCheerio.find (C:\Lab\Void\projects\mosaico\node_modules\cheerio\lib\api\traversing.js:55:30)
    at LoadedCheerio.Cheerio (C:\Lab\Void\projects\mosaico\node_modules\cheerio\lib\cheerio.js:82:30)
    at new LoadedCheerio (C:\Lab\Void\projects\mosaico\node_modules\cheerio\lib\load.js:32:46)
    at initialize (C:\Lab\Void\projects\mosaico\node_modules\cheerio\lib\load.js:38:16)
    at Object.<anonymous> (C:\Lab\Void\projects\mosaico\spec\converter-spec.js:48:15)
```

0.4.0 and 1.0.0-rc.1 to 1.0.0-rc.5 works fine.

I guess the problem is with the updated css-what dependency, but I don't understand why an html including only 2 comments should rise an exception in a css selector library.

**Comment by bago** [comment]
I can also confirm that 1.0.0-rc.5 with forced css-what 6.1.0 version works fine, so the issue is probably in cheerio (or another intermediate dependency, like cheerio-select) and not css-what.

**Comment by fb55** [comment]
Fixed in #2504
